src/rules/require-should-assertion-execution.js

```diff
@@ -5,15 +5,25 @@
  */
 
 const assert = require('assert');
-const should = require('should');
 
 /**
- * Auxiliary constants.
+ * Use `should` conditionally.
  */
 
-const Assertion = should.Assertion.prototype;
-const assertions = Object.keys(Assertion);
-const chains = Object.keys(Assertion).filter(key => typeof Assertion[key] !== 'function');
+let assertions;
+let chains;
+let should;
+
+try {
+  should = require('should');
+
+  const Assertion = should.Assertion.prototype;
+
+  assertions = Object.keys(Assertion);
+  chains = Object.keys(Assertion).filter(key => typeof Assertion[key] !== 'function');
+} catch (error) {
```

Comment:
This should be moved just after `should = require('should');` or the error should re-throw if code is not MODULE_NOT_FOUND.